// WekaServer/config/commonConfig.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.APP_NAME = "Weka";
exports.WELCOME_WORD = "Welcome to use Weka";
exports.LOG_FILE_EXPRESS = "logs/log_express.log";
exports.LOG_FILE = "logs/_app.log";
exports.DATE_PATTERN = "yyyy_MM_dd";
exports.ENV = "development"; //change this to production when go live
var MESSAGE_ANALYST_CONFIG = (function () {
    function MESSAGE_ANALYST_CONFIG() {
    }
    MESSAGE_ANALYST_CONFIG.getPattern = function (category) {
        for (var _i = 0, _a = MESSAGE_ANALYST_CONFIG.PATTERNS; _i < _a.length; _i++) {
            var pattern = _a[_i];
            if (pattern.category == category) {
                return pattern;
            }
        }
        return null;
    };
    return MESSAGE_ANALYST_CONFIG;
}());
MESSAGE_ANALYST_CONFIG.REGEX_SPLITTER = /[ $-/:-?{-~!"^_`\[\]|\u3002|\uff1f|\uff01|\uff0c|\u3001|\uff1b|\uff1a|\u201c|\u201d|\u2018|\u2019|\uff08|\uff09|\u300a|\u300b|\u3008|\u3009|\u3010|\u3011|\u300e|\u300f|\u300c|\u300d|\ufe43|\ufe44|\u3014|\u3015|\u2026|\u2014|\uff5e|\ufe4f|\uffe5]/;
MESSAGE_ANALYST_CONFIG.REGEX_MIXED = /[\u00ff-\uffff]|\S/g;
MESSAGE_ANALYST_CONFIG.REGEX_ENGLISH = /([A-Za-z])/g;
MESSAGE_ANALYST_CONFIG.REGEX_NUMBER = /([0-9])/g;
MESSAGE_ANALYST_CONFIG.REGEX_CHINESE = /[\u4e00-\u9fff\uf900-\ufaff]/g;
MESSAGE_ANALYST_CONFIG.REGEX_SYMBOL = /[$-/:-?{-~!"^_`\[\]|\u3002|\uff1f|\uff01|\uff0c|\u3001|\uff1b|\uff1a|\u201c|\u201d|\u2018|\u2019|\uff08|\uff09|\u300a|\u300b|\u3008|\u3009|\u3010|\u3011|\u300e|\u300f|\u300c|\u300d|\ufe43|\ufe44|\u3014|\u3015|\u2026|\u2014|\uff5e|\ufe4f|\uffe5]/g;
MESSAGE_ANALYST_CONFIG.PATTERNS = [{
        category: 0 /* Name */,
        countEnglish: { minimalLength: 0, maximalLength: 0, lowerLength: 0, higherLength: 0, weight: 1, },
        countNumber: { minimalLength: 0, maximalLength: 0, lowerLength: 0, higherLength: 0, weight: 1, },
        countChinese: { minimalLength: 0, maximalLength: 5, lowerLength: 2, higherLength: 4, weight: 10, },
        countSymbol: { minimalLength: 0, maximalLength: 0, lowerLength: 0, higherLength: 0, weight: 1, },
        keywords: "胡",
        keywordsWeight: 10,
        required: true,
        priority: 1,
    }, {
        category: 1 /* Address */,
        countEnglish: { minimalLength: 0, maximalLength: 0, lowerLength: 0, higherLength: 0, weight: 1, },
        countNumber: { minimalLength: 0, maximalLength: 0, lowerLength: 0, higherLength: 0, weight: 1, },
        countChinese: { minimalLength: 4, maximalLength: 40, lowerLength: 10, higherLength: 20, weight: 1, },
        countSymbol: { minimalLength: 0, maximalLength: 0, lowerLength: 0, higherLength: 0, weight: 1, },
        keywords: "省;县;区;镇;街道;社区;村;组;路;小区;号;幢;单元;楼;栋",
        keywordsWeight: 20,
        required: true,
        priority: 1,
    }, {
        category: 3 /* CommodityName */,
        countEnglish: { minimalLength: 0, maximalLength: 0, lowerLength: 0, higherLength: 0, weight: 1, },
        countNumber: { minimalLength: 0, maximalLength: 0, lowerLength: 0, higherLength: 0, weight: 1, },
        countChinese: { minimalLength: 4, maximalLength: 40, lowerLength: 10, higherLength: 20, weight: 1, },
        countSymbol: { minimalLength: 0, maximalLength: 0, lowerLength: 0, higherLength: 0, weight: 1, },
        keywords: "护肝片;维生素",
        keywordsWeight: 20,
        required: true,
        priority: 1,
    }, {
        category: 2 /* Mobile */,
        countEnglish: { minimalLength: 0, maximalLength: 0, lowerLength: 0, higherLength: 0, weight: 1, },
        countNumber: { minimalLength: 8, maximalLength: 20, lowerLength: 8, higherLength: 13, weight: 20, },
        countChinese: { minimalLength: 0, maximalLength: 0, lowerLength: 0, higherLength: 0, weight: 1, },
        countSymbol: { minimalLength: 0, maximalLength: 3, lowerLength: 0, higherLength: 1, weight: 1, },
        keywords: "",
        keywordsWeight: 0,
        required: true,
        priority: 1,
    }, {
        category: 4 /* Quantity */,
        countEnglish: { minimalLength: 0, maximalLength: 0, lowerLength: 0, higherLength: 0, weight: 1, },
        countNumber: { minimalLength: 0, maximalLength: 3, lowerLength: 1, higherLength: 2, weight: 10, },
        countChinese: { minimalLength: 0, maximalLength: 2, lowerLength: 0, higherLength: 2, weight: 10, },
        countSymbol: { minimalLength: 0, maximalLength: 3, lowerLength: 0, higherLength: 1, weight: 1, },
        keywords: "瓶;盒;箱;件;个;只;罐;条;袋;支;管",
        keywordsWeight: 10,
        required: true,
        priority: 1,
    },
];
exports.MESSAGE_ANALYST_CONFIG = MESSAGE_ANALYST_CONFIG;
