// WekaServer/server/logger.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var winston = require("winston");
require("winston-daily-rotate-file");
var commonConfiguration = require("../config/commonConfig");
var Logger = (function () {
    function Logger(label) {
        this.logger = new (winston.Logger)({
            transports: [
                new (winston.transports.Console)({
                    colorize: true,
                    prettyPrint: true,
                    timestamp: true,
                    label: label,
                }),
                new winston.transports.DailyRotateFile({
                    filename: commonConfiguration.LOG_FILE,
                    datePattern: commonConfiguration.DATE_PATTERN,
                    prepend: true,
                    label: label,
                    level: process.env.ENV === 'development' ? 'debug' : 'info'
                })
            ]
        });
    }
    Logger.prototype.info = function (logContent) {
        this.logger.log('info', logContent);
    };
    Logger.prototype.error = function (logContent) {
        this.logger.log('error', logContent);
    };
    Logger.prototype.debug = function (logContent) {
        this.logger.log('debug', logContent);
    };
    return Logger;
}());
exports.default = Logger;
